analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Roslyn/NullPointerDereference.CSharp8.cs

```diff
@@ -22,6 +25,11 @@ public void NullCoalescenceOperator_Null_Noncompliant2()
             name = name ?? name.ToString(); // Noncompliant
         }
 
+        public void NullCoalesce_Conversion(Exception arg)  // Classes doesn't make sense here. They just demonstrate the flow.
+        {
+            var value = arg as IDisposable ?? new FileStream(arg.Message, FileMode.Open);  // Noncompliant, if arg is null, it will throw
+        }
```

Comment:
I think it is more nuanced than that.
```suggestion

        public void NullCoalesce_Conversion_DownCast(ArgumentException arg)        {
            var value = arg as Exception ?? new FileStream(arg.Message, FileMode.Open);  // Noncompliant, arg must be null on the right side
        }

        public void NullCoalesce_Conversion_UpCast(Exception arg)        {
            var value = arg as ArgumentException ?? new FileStream(arg.Message, FileMode.Open);  // Compliant, arg can be null or another Exception type on the right side
        }        
```

Reply:
The 2nd says `Compliant. I guess it was a typo, as there could be null

Reply:
We don't know for sure that it is null. It likely isn't. So I would call it compliant. But that is debatable.